client/src/config.rs

```diff
@@ -78,17 +78,19 @@ impl From<&RuntimeConfig> for SyncClientConfig {
 	fn from(val: &RuntimeConfig) -> Self {
 		SyncClientConfig {
 			confidence: val.confidence,
-			disable_rpc: val.disable_rpc,
+			disable_rpc: matches!(val.network_mode, NetworkMode::P2POnly),
 			app_id: val.app_id,
+			network_mode: val.network_mode,
 		}
 	}
 }
 
 impl From<&RuntimeConfig> for AppClientConfig {
 	fn from(val: &RuntimeConfig) -> Self {
 		AppClientConfig {
-			disable_rpc: val.disable_rpc,
+			disable_rpc: matches!(val.network_mode, NetworkMode::P2POnly),
 			threshold: val.threshold,
+			network_mode: val.network_mode,
```

Comment:
We can remove `disable_rpc` entirely and use equality directly in the code, since we are storing network_mode in the configuration.